woocommerce/woocommerce
# [e2e tests] Improve execution time by disabling the welcomeGuide feature instead of waiting for the modal

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Every time the page or the post editor is visited the `closeWelcomeModal` function waits 5 seconds for the welcome tour modal to be displayed and closes it. Most of the times the modal is not even being displayed, as it was previously dismissed.

This PR adds a function to disable the modal by setting the `welcomeGuide` feature. It should save 5 seconds of most editor page visits.

I also removed some unnecessary `waitFor('networkidle')` calls that were slowing down the tests.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Run all affected tests, they should pass.

`pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/create-cart-block.spec.js ./tests/e2e-pw/tests/shopper/cart-block-calculate-shipping.spec.js ./tests/e2e-pw/tests/shopper/cart-block-coupons.spec.js ./tests/e2e-pw/tests/shopper/cart-block.spec.js ./tests/shopper/cart-checkout-block-calculate-tax.spec.js ./tests/e2e-pw/tests/shopper/mini-cart.spec.js ./tests/e2e-pw/tests/shopper/shop-products-filter-by-price.spec.js, ./tests/e2e-pw/tests/merchant/create-post.spec.js, ./tests/e2e-pw/tests/merchant/create-page.spec.js, ./tests/e2e-pw/tests/shopper/product-tags-attributes.spec.js`



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
